index.bs

```diff
@@ -365,76 +380,123 @@ spec: webidl
   </section>
 </section>
 <section>
-  <h2 dfn-type="dfn" export>
-    Permission Store
-  </h2>
+  <h2 id="permission-stores">Permission stores</h2>
   <p>
-    <a lt='user agent'>User agents</a> MAY use a form of storage to
-    keep track of web site permissions. When they do, they MUST have a
-    <dfn export>permission storage identifier</dfn> which is linked to a
-    {{PermissionStorage}} instance or one of its subtypes.
+    Each <a>realm</a> has a <dfn export>permission store</dfn> consisting of,
+    for each {{PermissionName}}, either a <dfn>denied flag</dfn> or a set
+    of <a>permission entries</a>. <dfn>Permission entries</dfn> for a
+    {{PermissionName}} are instances of that {{PermissionName}}'s <a>permission
+    entry type</a>, which will be {{PermissionStorageEntry}} or one of its
+    subtypes. The <a>permission entry type</a> specifies the <a>equivalent
+    permission entry</a> relation used to de-duplicate the <a>permission
+    entries</a>.
   </p>
+
   <p>
-    To <dfn>get a permission storage identifier</dfn> for a
-    {{PermissionName}} <var>name</var> and an <a>environment settings
-    object</a> <var>settings</var>, the UA MUST return a tuple consisting
-    of:
+    The UA may initialize <a>realm</a>'s <a>permission store</a> almost
+    arbitrarily, subject to the constraints in <a
+    href="#permission-store-constraints"></a>. The UA may also <a>queue a
+    task</a> to update a <a>realm's</a> <a>permission store</a> almost
+    arbitrarily, also subject to the constraints in <a
+    href="#permission-store-constraints"></a>.
   </p>
-  <ol>
-    <li>
-      <var>name</var>
-    </li>
-    <li>
-      <var>settings</var>' <a>origin</a>
-    </li>
-    <li>optional UA-specific data like whether <var>settings</var>'
-    <a>browsing context</a> has a <a>parent browsing context</a>, or
-    <var>settings</var>' <a>top-level browsing context</a>'s <a>origin</a>
-    </li>
-  </ol>
+
+  <p>
+    The UA may store <a>permission entries</a> more persistently than a
+    <a>realm</a>, but this specification doesn't constrain how.
+  </p>
```

Comment:
It should at least be constrained to origins. Cannot allow new kind of cookies. And ideally we do make it possible for certain permissions to require something more here. E.g., persistent storage isn't really useful without origin-wide permission.